### PR TITLE
cmake: Disable eager MSan in ctime_tests

### PR DESCRIPTION
**Pull request by hebasto**
Same as https://github.com/bitcoin-core/secp256k1/pull/1517, but for the CMakle build system.

The second commit improves the configure summary (similar to https://github.com/hebasto/bitcoin/pull/189.

### CODE REVIEW

**Review by real-or-random — COMMENTED** [review]
Concept ACK

File: cmake/CheckMemorySanitizer.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+include_guard(GLOBAL)
+include(CheckCSourceCompiles)
+
+function(check_memory_sanitizer output)
+  set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+  check_c_source_compiles("
+    #if defined(__has_feature)
+    #  if __has_feature(memory_sanitizer)
+    //   MemorySanitizer is enabled.
```
Comment (real-or-random) [inline cmake/CheckMemorySanitizer.cmake:9]: nit: We should probably use the same test program on configure and CMake (and perhaps avoid `//` comments which are invalid in C89, though that really is more of a philosophical convern.)

File: CMakeLists.txt [hunk 2]
```diff
@@ -263,6 +263,17 @@ endif()
 
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 
+set(print_msan_notice)
+if(SECP256K1_BUILD_CTIME_TESTS AND CMAKE_C_COMPILER_ID MATCHES "^(Clang|AppleClang)$")
```
Comment (fanquake) [inline CMakeLists.txt:267]: Does Apple Clang makes sense here?

Comment (fanquake) [reply CMakeLists.txt:267]: Actually, why are you hard coding compiler IDs at all? This just makes things less flexible, and the check should be enough.

Comment (real-or-random) [reply CMakeLists.txt:267]: Yeah, I had the same thought. It saves a test compilation for GCC, but it's probably simpler to drop the check for clang. In https://github.com/bitcoin-core/secp256k1/pull/1517, I also added a similar check (but for "GCC or Clang", so it's much more useless). Let's just drop it.

**Comment by hebasto** [comment]
Addressed comments:
- https://github.com/bitcoin-core/secp256k1/pull/1532#discussion_r1615986643:
> avoid `//` comments which are invalid in C89

- https://github.com/bitcoin-core/secp256k1/pull/1532#discussion_r1615994037
> Actually, why are you hard coding compiler IDs at all? This just makes things less flexible, and the check should be enough.

File: cmake/CheckMemorySanitizer.cmake (hunk 1 above)
Comment (hebasto) [reply cmake/CheckMemorySanitizer.cmake]: > nit: We should probably use the same test program on configure and CMake

I'd prefer align Autotools' `SECP_MSAN_CHECK` with CMake's `check_memory_sanitizer` to keep the latter straightforward without negated values.

> ... and perhaps avoid `//` comments which are invalid in C89

Thanks. [Fixed](https://github.com/bitcoin-core/secp256k1/pull/1532#issuecomment-2133405610).

File: CMakeLists.txt (hunk 2 above)
Comment (hebasto) [reply CMakeLists.txt]: > Actually, why are you hard coding compiler IDs at all? This just makes things less flexible, and the check should be enough.

Thanks! The compiler ID check has been [deleted](https://github.com/bitcoin-core/secp256k1/pull/1532#issuecomment-2133405610).

File: cmake/CheckMemorySanitizer.cmake (hunk 1 above)
Comment (real-or-random) [reply cmake/CheckMemorySanitizer.cmake]: > I'd prefer to align Autotools' `SECP_MSAN_CHECK` code with CMake's `check_memory_sanitizer` to keep the latter straightforward without negated values.

Fine with that if you want to add a commit that makes them consistent, but then we'll also need to `#error` if `!defined(__has_feature)`.

File: CMakeLists.txt (hunk 2 above)
Comment (real-or-random) [reply CMakeLists.txt]: Thanks, do you want to add a commit that also removes the `$GCC` check in configure.ac, just to keep the system consistent?

**Comment by hebasto** [comment]
Addressed comments:
- https://github.com/bitcoin-core/secp256k1/pull/1532#discussion_r1616380720:
> ... we'll also need to `#error` if `!defined(__has_feature)`.

- https://github.com/bitcoin-core/secp256k1/pull/1532#discussion_r1616382346:
> ... to add a commit that also removes the $GCC check in configure.ac, just to keep the system consistent

File: cmake/CheckMemorySanitizer.cmake (hunk 1 above)
Comment (hebasto) [reply cmake/CheckMemorySanitizer.cmake]: > ... but then we'll also need to `#error` if `!defined(__has_feature)`.

Thanks! [Fixed](https://github.com/bitcoin-core/secp256k1/pull/1532#issuecomment-2134674954).

File: CMakeLists.txt (hunk 2 above)
Comment (hebasto) [reply CMakeLists.txt]: > Thanks, do you want to add a commit that also removes the `$GCC` check in configure.ac, just to keep the system consistent?

Sure. [Added](https://github.com/bitcoin-core/secp256k1/pull/1532#issuecomment-2134674954).

**Review by real-or-random — APPROVED** [review]
ACK f55703ba49454fc46226f4846fe292d4a3dfa3ef

**Pull request merged**